client/src/pages/TaskFormPage/TaskFormPage.js

```diff
@@ -1,40 +1,138 @@
+    }
+
+    const startDate = new Date(taskFormData.startDate);
+    const endDate = new Date(taskFormData.endDate);
+    if (
+      startDate.getTime() > endDate.getTime() ||
+      startDate.getTime() < new Date().getTime()
+    ) {
+      return "Start time must come before end time and after the current time";
+    }
+
+    return ""; // empty string means successful validation
+  };
+
+  const createTaskHandler = async () => {
+    setErrorMessage("");
+    const validationResult = validateTaskFormData();
+
+    if (validationResult !== "") {
+      setErrorMessage(validationResult);
+      console.log(validationResult);
+      return;
+    }
+
+    const taskFormUploadData = { ...taskFormData };
+    taskFormUploadData.startDate = taskFormUploadData.startDate.toISOString();
+    taskFormUploadData.endDate = taskFormUploadData.endDate.toISOString();
+
+    console.log(taskFormUploadData);
+
+    await instance
+      .post("/api/tasks", taskFormUploadData)
+      .then((res) => {
+        console.log(`Response: ${res.data}`);
+        routeToCalendar();
+      })
+      .catch((e) => {
+        console.log(e);
+        setErrorMessage("Request Failed");
+      });
+  };
+
   return (
     <div className={styles.bg}>
       <div>
-        <button
-          type={"button"}
-          className={styles.closeButton}
-          aria-label={"Close"}
-        >
-          &times;
-        </button>
         <p className={styles.header}>Create a Task</p>
         <div className={styles.inputBox}>
           <InputBox
             header={"Task Title"}
             type={"text"}
             placeholder={"Task Name"}
+            onChange={(e) => updateTaskFormData(e, "title")}
+            value={taskFormData.title}
           />
           <InputBox
             header={"Location"}
             type={"text"}
             placeholder={"Location Name"}
+            onChange={(e) => updateTaskFormData(e, "location")}
+            value={taskFormData.location}
           />
           <InputBox
             header={"Description"}
             type={"text"}
             placeholder={""}
             expanded={true}
+            onChange={(e) => updateTaskFormData(e, "description")}
+            value={taskFormData.description}
+          />
+          <p className={styles.inputHeader}>Start</p>
+          <DateSelector
+            showTime={true}
+            selectedDate={taskFormData.startDate}
+            onDateChanged={(newDate) =>
+              updateTaskFormData(newDate, "startDate")
+            }
+          />
+
+          <p className={styles.inputHeader}>End</p>
+          <DateSelector
+            showTime={true}
+            selectedDate={taskFormData.endDate}
+            onDateChanged={(newDate) => updateTaskFormData(newDate, "endDate")}
           />
         </div>
-        <div className={styles.button}>
-          <CredentialsButton text={"Create"} />
+
+        <div className={styles.buttonBar}>
```

Comment:
Minor comment: I would just swap the places of the Create and Cancel button (It seems more fitting to have create be on the left rather than on the right so it's flush with the rest of the Create Task Form)

Reply:
Minor UI Comment: Also adding some hover styles (i.e. Changing the background to be a bit darker) might also be worth considering

Reply:
Will check with the team for "Cancel" and "Create" position.

Will put the hover style improvement on hold.

Reply:
Noted